### PR TITLE
refactor: reorganize and improve DotSteps Expandable and Icon stories

### PR DESCRIPTION
**Pull request by baptisteArno**

### CODE REVIEW

File: src/components/Icon/__stories__/Name.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import Icon, { icons } from '..'
+import Stack from '../../Stack'
+
+const StyledStack = styled(Stack)`
+  font-size: ${({ theme }) => theme.typography.heading.fontSize};
+`
+
+export const Name: ComponentStory<typeof Icon> = args => (
+  <>
+    {icons.map(name => (
```
Comment (QuiiBz) [inline src/components/Icon/__stories__/Name.stories.tsx:12]: Can we remove this fragment, like in other stories?

File: src/components/Icon/__stories__/Size.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import { DecoratorFunction } from '@storybook/addons'
+import { ComponentStory } from '@storybook/react'
+import { ComponentProps } from 'react'
+import Icon from '..'
+
+const sizes: ComponentProps<typeof Icon>['size'][] = [40, 50, 60]
+
+export const Size: ComponentStory<typeof Icon> = args => (
+  <>
+    {sizes.map(size => (
```
Comment (QuiiBz) [inline src/components/Icon/__stories__/Size.stories.tsx:10]: Same for the fragment

File: src/components/Icon/__stories__/Name.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+      <StyledStack key={name} direction="row" gap={2} alignItems="center">
+        <Icon name={name} {...args} /> {name}
+      </StyledStack>
+    ))}
+  </>
+)
```
Comment (matthprost) [inline src/components/Icon/__stories__/Name.stories.tsx:18]: it is not very beautiful, can't we just have icons with stack as parent?

Screenshot 2022-11-08 at 14 18 09

**Review by matthprost — CHANGES_REQUESTED** [review]

File: src/components/Expandable/__stories__/Controlled.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import { DecoratorFunction } from '@storybook/addons'
+import { useState } from 'react'
+import Expandable from '..'
+import Button from '../../Button'
+import Stack from '../../Stack'
+
+export const Controlled = () => {
```
Comment (DorianMaliszewski) [inline src/components/Expandable/__stories__/Controlled.stories.tsx:7]:
```suggestion
export const Controlled: Story = () => {
```

Then you can drop your as DecoratorFunction below ?

File: src/components/Icon/__stories__/Color.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+import { ComponentProps } from 'react'
+import Icon from '..'
+
+const colors: ComponentProps<typeof Icon>['color'][] = [
+  'primary',
+  'success',
+  'warning',
+]
+
+export const Color = (args: ComponentProps<typeof Icon>) =>
```
Comment (DorianMaliszewski) [inline src/components/Icon/__stories__/Color.stories.tsx:11]:
```suggestion
export const Color: ComponentStory<typeof Icon> = (args) =>
```

File: src/components/Icon/__stories__/Color.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+Color.parameters = {
+  docs: {
+    storyDescription: 'Set size using `size` property.',
+  },
+}
+
+Color.decorators = [
+  Story => (
+    <div style={{ alignItems: 'center', display: 'flex', gap: 16 }}>
```
Comment (DorianMaliszewski) [inline src/components/Icon/__stories__/Color.stories.tsx:22]: Use a Stack ?

**Review by QuiiBz — APPROVED** [review]

**Review by matthprost — APPROVED** [review]

**Pull request merged**